tc39_test.go

```diff
@@ -525,7 +711,72 @@ func (ctx *tc39TestCtx) runTC39Test(name, src string, meta *tc39Meta, t testing.
 	vm.Set("IgnorableTestError", ignorableTestError)
 	vm.RunProgram(ctx.sabStub)
 	var out []string
+	eventLoopQueue := make(chan func(), 2) // the most basic and likely buggy event loop
 	async := meta.hasFlag("async")
+
+	type cacheElement struct {
+		m   ModuleRecord
+		err error
+	}
+	cache := make(map[string]cacheElement)
+	mx := sync.Mutex{}
+
+	var hostResolveImportedModule func(referencingScriptOrModule interface{}, specifier string) (ModuleRecord, error)
+	hostResolveImportedModule = func(referencingScriptOrModule interface{}, specifier string) (ModuleRecord, error) {
+		mx.Lock()
+		defer mx.Unlock()
+		fname := path.Join(ctx.base, path.Dir(name), specifier)
```

Comment:
This likely needs more love, I basically did the bare minimum to make it have modules support,and then I needed "event loop" for the dynamic import. I guess I could've implemented it without an "event loop" if the dynamic import hooks weren't asynchronous, but that seemed like the better way to do it. I currently can't remember if there was a particular test that wouldn't pass otherwise, or if there was any other kind of problem with this idea.

I will try to move some of the inline lambdas outside and maybe have a pass at trying to make some of the functions smaller, but in general I do not want to make a whole event loop. If a real event loop is create for this it likely will be better idea to also make it part of some package to test it 🤷‍♂. I guess we can use `goja_nodejs` but I don't know how that will work with go modules and whether that is a good idea 🤷‍♂